File: Message/src/main/java/com/yahoo/sdvornik/message/StartSortingMessage.java
package com.yahoo.sdvornik.message;

import io.netty.buffer.ByteBuf;

public class StartSortingMessage  extends Message {
    private int content;

    public int getContent() {
        return content;
    }

    public StartSortingMessage(int content) {
        this.content = content;
    }

    @Override
    public Type getType() {
        return Type.START_SORTING;
    }

    @Override
    public long getLength() {
        return 2*Integer.BYTES;
    }

    @Override
    public void writeContent(ByteBuf buf) {
        buf.writeInt(content);
    }
}

File: shared/src/main/kotlin/com/yahoo/sdvornik/message/StartSortingMessage.kt
package com.yahoo.sdvornik.message

import io.netty.buffer.ByteBuf

/**
 * @author Serg Dvornik <sdvornik@yahoo.com>
 */
class StartSortingMessage(val content: Int) : Message() {

    override val type: Message.Type = Message.Type.START_SORTING

    override val length: Long= 2 * Integer.BYTES.toLong()

    override fun writeContent(buf: ByteBuf) {
        buf.writeInt(content)
    }
}